programmer-skills/http-intro.md

```diff
@@ -0,0 +1,361 @@
+Introduction to HTTP
+====================
+
+The Hypertext Transfer Protocol (HTTP) is the foundation of web communication, so some familiarity with it is essential for any web developer. HTTP works on the basis of a *client* sending a *request* to a *server*, which in return provides a *response*. You can think of the client as being the user's web browser and the server as being any kind of back-end server you might write (e.g. in Express.js).
```

Comment:
Reads better "Hypertext Transfer Protocol (HTTP) is one of the main foundations of web communication.  As a developer it is prudent to be familiar with it.  HTTP works on the basis of a *client* sending a *request* to a *server*, which in return provides a *response*.  For example, a web browser is a client program at the user computer that may access information at any web server in the world.